### PR TITLE
Support for Glide 4.0

### PR DESCRIPTION
**Pull request by samtstern**
See #731

### CODE REVIEW

File: storage/src/main/java/com/firebase/ui/storage/images/FirebaseImageLoader.java [hunk 1]
```diff
@@ -1,38 +1,100 @@
...
+                new FirebaseStorageKey(reference),
+                new FirebaseStorageFetcher(reference));
+    }
+
+    @Override
+    public boolean handles(StorageReference reference) {
+        return true;
+    }
+
+    private class FirebaseStorageKey implements Key {
```
Comment (TheCraftKid) [inline storage/src/main/java/com/firebase/ui/storage/images/FirebaseImageLoader.java]: Should this be static?

Comment (samtstern) [inline storage/src/main/java/com/firebase/ui/storage/images/FirebaseImageLoader.java]: Good call.

**Comment by Sourav-21** [comment]
Change
```
public void registerComponents(Context context, Registry registry) {
        registry.append(StorageReference.class, InputStream.class, new FirebaseImageLoader.Factory());
    }
```
to 
```
public void registerComponents(Context context, Glide glide, Registry registry) {
        registry.append(StorageReference.class, InputStream.class, new FirebaseImageLoader.Factory());
    }
```

**Comment by samtstern** [comment]
@SUPERCILEX @TheCraftKid any objections to pushing this through for `3.0`?

**Comment by samtstern** [comment]
@Sourav-21 can you explain your comment? That does not seem to be a valid override on `AppGlideModule`.

**Comment by TheCraftKid** [comment]
LGTM!

**Comment by Sourav-21** [comment]
Using `GlideApp` instead of `Glide` will be better.

**Comment by TheCraftKid** [comment]
@Sourav-21 Is there a specific reason? The samples don't make that distinction.

**Comment by Sourav-21** [comment]
With the generated API, the `RequestOptions` calls can be inlined:

```
GlideApp.with(fragment)
    .load(url)
    .centerCrop()
    .placeholder(R.drawable.placeholder)
    .error(R.drawable.error)
    .priority(Priority.HIGH)
    .into(imageView);
```

**Comment by SUPERCILEX** [comment]
@Sourav-21 I don't think you can add new `load()` methods with the generated API. In addition, you'll still be using it on your side—we just provide the loading mechanism.

@samtstern I'm not super familiar with providing loaders for Glide, but at a glance this LGTM. 😄

**Comment by Sourav-21** [comment]
@SUPERCILEX I am just giving an example. `load(storageRef)` method with the generated API working fine.

**Comment by Sourav-21** [comment]
With the release of Glide 4.1.1. I suggest the following changes
update `Glide` dependencies to 4.1.1 is 
`provided 'com.github.bumptech.glide:glide:4.1.1'`

Change the `GlideModule`
```
@GlideModule
public class MyGlideModule extends AppGlideModule {
    @Override
    public void registerComponents(Context context, Glide glide, Registry registry) {
        registry.append(StorageReference.class, InputStream.class, new FirebaseImageLoader.Factory());
    }
}
```
And use `GlideApp` instead of `Glide` for simplicity.
```
GlideApp.with(fragment)
    .load(storageRef)
    .centerCrop()
    .placeholder(R.drawable.placeholder)
    .error(R.drawable.error)
    .priority(Priority.HIGH)
    .into(imageView);
```

**Comment by SUPERCILEX** [comment]
Ohhhhh, I get it. You're suggesting simplifying the sample. Yeah @samtstern that's probably a good idea.

**Comment by Sourav-21** [comment]
Yeah, I am suggesting you modify the demo. The changes the changes @TheCraftKid has done with `FirebaseImageLoader.java` is great.

**Comment by samtstern** [comment]
@Sourav-21 thanks for the pointer about `GlideApp`.

I think this is ready to go now.

**Pull request merged**